src/js/components/TaskTable.js

```diff
@@ -165,6 +184,12 @@ class TaskTable extends React.Component {
     );
   }
 
+  renderHost(prop, task) {
+    let marathonTask = MarathonStore.getTaskFromTaskID(task.id);
```

Comment:
actually, the fact that we pick out the task here is extremely expensive. if we have 500 tasks (which is quite common) then we're going to be iterating through a shit load of things. 

here's the implementation of `getTaskFromTaskID` https://github.com/dcos/dcos-ui/blob/master/src/js/stores/MarathonStore.js#L343

i'd say we should definitely find a way around this somehow before this merges. no idea how